anime_downloader/extractors/gcloud.py

```diff
@@ -5,27 +5,96 @@
     def _get_data(self):
         url = self.url
         
-        """gcloud uses the same video ID as other sites"""
-        url = url.replace('fembed.com','gcloud.live')
-        url = url.replace('feurl.com','gcloud.live')
+        extVer = None
+        if not hasattr(self, 'extractorVersion'):
+            setattr(self, 'extractorVersion', 1)            
+        extVer = getattr(self, 'extractorVersion')
+        logger.debug( "extVer: {}".format(extVer))
         
-        url = url.replace('gcloud.live/v/','gcloud.live/api/source/')
-        if url.find('#') != -1:url = url[:url.find('#')] 
-        url = (url[-url[::-1].find('/'):])
-        data = helpers.post(f'https://gcloud.live/api/source/{url}').json()['data']
+        fallback_qualities = None
+        if not hasattr(self, 'fallback_qualities'):
+            setattr(self, 'fallback_qualities', None)            
+        fallback_qualities = getattr(self, 'fallback_qualities')
+        logger.debug( "fallback_qualities: {}".format(fallback_qualities))
+
 
-        if data == 'Video not found or has been removed':
-            logger.warning('File not found (Most likely deleted)')
-            return {'stream_url': ''}
+        # Reinit to V1 for next Call by an another site config
+        #setattr(self, 'extractorVersion', 1)
+        
+        if extVer == 1:
+            """gcloud uses the same video ID as other sites"""
+            url = url.replace('fembed.com','gcloud.live')
+            url = url.replace('feurl.com','gcloud.live')
+            
+            url = url.replace('gcloud.live/v/','gcloud.live/api/source/')
+            if url.find('#') != -1:url = url[:url.find('#')] 
+            url = (url[-url[::-1].find('/'):])
+            data = helpers.post(f'https://gcloud.live/api/source/{url}').json()['data']
+    
+            if data == 'Video not found or has been removed':
+                logger.warning('File not found (Most likely deleted)')
+                return {'stream_url': ''}
+            
+            for a in data:
+                if a['label'] == self.quality:
+                    return {'stream_url': a['file']}
         
-        for a in data:
-            if a['label'] == self.quality:
-                return {'stream_url': a['file']}
+        else:
+            if url.find('#') != -1:url = url[:url.find('#')]
+            urlapi = url.replace('/v/', '/api/source/' )
+            
+            data = None
+            try:
+                data = helpers.post(urlapi, referer=url).json()
+            except HTTPError as err:
+                logger.debug("_get_data HTTP Err: %s", str(err))
+                
+            if data:
+                #logger.debug( "data: {}".format(data) )
+                logger.debug( 'Success: %s', str(data['success']) )
+                logger.debug( "data: {}".format(data['data']) )
+                
+                url_quality = {}
+                
+                for d in data['data']:
```

Comment:
Something like this would be better

```
# d['label'] is the quality of stream
dict((d['label'], d['file'] for d in data['data'] if d['label'] in url_quality))
```